CentralComputing/UDPManager.cpp

```diff
@@ -77,38 +77,37 @@ bool UDPManager::start_udp(const char * hostname, const char * send_port, const
   return false;
 }
 
-void UDPManager::udp_recv(){
-  uint8_t recv_buf[16];
-  socklen_t fromlen; // needed for recvfrom
-  struct sockaddr_storage fromaddr;// needed for recvfrom
-  fromlen = sizeof fromaddr; // needed for recvfrom
+int UDPManager::udp_recv(uint8_t* recv_buf, uint8_t len){
+  
+  socklen_t fromlen;                // needed for recvfrom
+  struct sockaddr_storage fromaddr; // needed for recvfrom
+  fromlen = sizeof fromaddr;        // needed for recvfrom
 
-  int byte_count = recvfrom(socketfd, recv_buf, sizeof recv_buf, 0, 
+  int byte_count = recvfrom(socketfd, recv_buf, len, 0, 
       (struct sockaddr *)&fromaddr, &fromlen);
-  //int byte_count = recv(socketfd, recv_buf, sizeof recv_buf,0);
 
   recv_buf[byte_count] = '\0';
   print(LogLevel::LOG_DEBUG, "recv %d bytes, they are: %s \n", byte_count, recv_buf);
+  
+  udp_parse(recv_buf, len);
```

Comment:
Move the `udp_parse()` function outside of `udp_recv()`. 
This way `udp_recv()` only does one thing, and can be tested that way. It is generally better practice to isolate behavior